tests/test_foundation/test_network_types.py

```diff
@@ -15,3 +15,45 @@ def test_GenTLV_value_unpack(self):
         tlv_unpacked = GenericTLV()
         tlv_unpacked.unpack(tlv.pack())
         self.assertEqual(tlv.value.value, tlv_unpacked.value.value)
+
+
+class TestIPv4(unittest.TestCase):
+    """Test IPv4 packets."""
+
+    def test_IPv4_pack(self):
+        """Test pack/unpack of IPv4 class."""
+        packet = IPv4(dscp=10, ttl=64, protocol=17, source="192.168.0.10",
+                      destination="172.16.10.30", options=b'1000',
+                      data=b'testdata')
+        packed = packet.pack()
+        expected = b'F(\x00 \x00\x00\x00\x00@\x11\x02'
+        expected += b'\xc5\xc0\xa8\x00\n\xac\x10\n\x1e1000testdata'
+        self.assertEqual(packed, expected)
+
+    def test_IPv4_unpack(self):
+        """Test unpack of IPv4 binary packet."""
+        raw = b'FP\x00$\x00\x00\x00\x00\x80\x06W'
+        raw += b'\xf4\n\x9aN\x81\xc0\xa8\xc7\xcc1000somemoredata'
+        expected = IPv4(dscp=20, ttl=128, protocol=6, source="10.154.78.129",
+                        destination="192.168.199.204", options=b'1000',
+                        data=b'somemoredata')
+        expected.pack()
```

Comment:
Yes it does: the pack() method calculates the length, IHL and checksum for the packet.